app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt

```diff
@@ -1,5 +1,130 @@
 package org.p2p.wallet.history.ui.history
 
+import kotlinx.coroutines.launch
+import org.p2p.wallet.common.analytics.interactor.ScreensAnalyticsInteractor
 import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.common.ui.recycler.PagingState
+import org.p2p.wallet.history.interactor.HistoryInteractor
+import org.p2p.wallet.history.model.HistoryTransaction
+import org.p2p.wallet.infrastructure.network.data.EmptyDataException
+import org.p2p.wallet.infrastructure.network.provider.TokenKeyProvider
+import org.p2p.wallet.receive.analytics.ReceiveAnalytics
+import org.p2p.wallet.renbtc.interactor.RenBtcInteractor
+import org.p2p.wallet.send.analytics.SendAnalytics
+import org.p2p.wallet.swap.analytics.SwapAnalytics
+import timber.log.Timber
+import java.math.BigDecimal
+import kotlin.properties.Delegates
 
-class HistoryPresenter : BasePresenter<HistoryContract.View>(), HistoryContract.Presenter
+private const val PAGE_SIZE = 20
+
+class HistoryPresenter(
+    private val historyInteractor: HistoryInteractor,
+    private val tokenKeyProvider: TokenKeyProvider,
+    private val renBtcInteractor: RenBtcInteractor,
+    private val receiveAnalytics: ReceiveAnalytics,
+    private val swapAnalytics: SwapAnalytics,
+    private val analyticsInteractor: ScreensAnalyticsInteractor,
+    private val sendAnalytics: SendAnalytics
+) : BasePresenter<HistoryContract.View>(), HistoryContract.Presenter {
+
+    private val publicKey = tokenKeyProvider.publicKey
```

Comment:
who's public key is that? User's or ...?

Reply:
When you pass registration you create SOL Account so public key it's SOL public key